code/modules/projectiles/projectile.dm

```diff
@@ -63,14 +63,26 @@
 /obj/item/projectile/proc/on_range() //if we want there to be effects when they reach the end of their range
 	qdel(src)
 
-/obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)

```

Comment:
Removed the hit_zone argument, this is exactly the same as the def_zone var of the projectile.